src/analyses/mutexAnalysis.ml

```diff
@@ -138,12 +149,39 @@ struct
               | Spawn -> false (* TODO: nonsense? *)
             in
             let el = P.effect_fun ~write ls in
-            ctx.sideg v el
+            ctx.sideg v el;
+
+            if !GU.postsolving && GobConfig.get_bool "dbg.print_protection" then (
+              let held_locks = P.export ~write (ctx.global v) in
+              let gm = if write then gm_w else gm_rw in
+              let vs_empty = VarSet.empty () in
+              Mutexes.iter (fun addr ->
+                  GM.modify_def vs_empty addr (VarSet.add v) gm
+                ) held_locks
+            )
         | None -> M.info ~category:Unsound "Write to unknown address: privatization is unsound."
       end;
       ctx.local
     | _ ->
       event ctx e octx (* delegate to must lockset analysis *)
+
+  let finalize () =
+    if GobConfig.get_bool "dbg.print_protection" then (
+      let max_cluster = ref 0 in
+      let num_mutexes = ref 0 in
+      let sum_protected = ref 0 in
+      Printf.printf "\n\nProtecting mutexes:\n";
+      (* TODO: what about rw? *)
```

Comment:
This logging output is currently about just write-protection, which is what apron privatizations cared about, but maybe it should be about read-write-protection instead?
Actually, in apron this ignored the write flag, so was a mish-mash of both anyway.

Reply:
Makes sense!